data_structures/heap/min_heap.py

```diff
@@ -0,0 +1,140 @@
+# Min head data structure
+# with decrease key functionality - in O(log(n)) time
+
+
+class MinHeap:
+    def __init__(self, array):
+        self.idx_of_element = {}
+        self.heap = self.buildHeap(array)
+
+    def getParentIdx(self, idx):
+        return (idx - 1) // 2
+
+    def getLeftChildIdx(self, idx):
+        return idx * 2 + 1
+
+    def getRightChildIdx(self, idx):
+        return idx * 2 + 2
+
+    def buildHeap(self, array):
+        lastIdx = len(array) - 1
+        startFrom = self.getParentIdx(lastIdx)
+
+        for idx, i in enumerate(array):
+            self.idx_of_element[i] = idx
+
+        for i in range(startFrom, -1, -1):
+            self.siftDown(i, array)
+        return array
+
+    # this is min-heapify method
+    def siftDown(self, idx, array):
```

Comment:
sift_down()

Reply:
Ok, I'll change names.

Reply:
Shall I change all method names to `sift_down()` convention or only this method?

Reply:
I have changed the names and also added doctest

Reply:
In Python the convention is that functions and methods use_lowercase_names() while classes UseUppercaseNames().

Reply:
Changed